pkg/views/gitprovider/select.go

```diff
@@ -106,6 +111,43 @@ func GitProviderSelectionView(ctx context.Context, gitProviderAddView *apiclient
 				Description("Will default to username if left empty").
 				Value(gitProviderAddView.Alias),
 		).WithHeight(6),
+
+		huh.NewGroup(huh.NewSelect[string]().
+			Title("Commit Signing Method").
+			DescriptionFunc(func() string {
+				return getGitProviderSigningHelpMessage(*gitProviderAddView.Id)
+			}, nil).
+			Options(
+				huh.Option[string]{Key: "None", Value: "none"},
+				huh.Option[string]{Key: "SSH", Value: "ssh"},
+				huh.Option[string]{Key: "GPG", Value: "gpg"},
+			).
+			Value(&selectedSigningMethod).WithHeight(6),
+		).WithHeight(8).WithHideFunc(func() bool {
+			return commitSigningNotSupported(*gitProviderAddView.Id)
```

Comment:
Having a nil pointer exception here when trying to add git provider (any of them throws exception). Let's tend to this issue.
You probably haven't experienced it due to having git providers already registered. 
Advice for final testings in future, run `daytona purge` and test fresh from the scratch. That way it's easier to catch issues like this. 😄 
 

https://github.com/user-attachments/assets/e3c538bd-1dd6-40fd-98d4-43fde15e9208

Reply:
@lbrecic Please give me some time. The merge conflicts had changed the structure of code. therefore I will need to retest everything. I will inform you when I'm done
PS: Sorry for late reply

Reply:
@bryans-go alright, take your time, I know there were a lot of conflicts. My advice is to retest everything with fresh daytona as I mentioned above, just to be sure we covered everything. Reach out when you think code is ready for me to test it. :)

Reply:
[![asciicast](https://asciinema.org/a/HrZc89hHq7t5HlwT6Je8w2WtL.svg)](https://asciinema.org/a/HrZc89hHq7t5HlwT6Je8w2WtL)

Reply:
@lbrecic  I have tested the signing and also attached a video for reference.